// vite-plugin-checker/dist/checkers/tscUtils.js
function forceNoEmitOnSolutionBuilderHost(ts, host) {
  var _a;
  const parseConfigHost = {
    useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames,
    readDirectory: ts.sys.readDirectory,
    fileExists: ts.sys.fileExists,
    readFile: ts.sys.readFile,
    getCurrentDirectory: ts.sys.getCurrentDirectory,
    onUnRecoverableConfigFileDiagnostic: () => {
    }
  };
  const original = (_a = host.getParsedCommandLine) == null ? void 0 : _a.bind(host);
  host.getParsedCommandLine = (fileName) => {
    const parsed = original ? original(fileName) : ts.getParsedCommandLineOfConfigFile(
      fileName,
      // `tscBuild` is the internal flag the solution builder sets on every
      // referenced project's parsed options. Without it `canEmitTsBuildInfo`
      // returns false, so a `noEmit: true` referenced project whose
      // `include` resolves to only `.d.ts` files has no output path at all
      // and `getFirstProjectOutput` crashes. 
      { tscBuild: true },
      parseConfigHost
    );
    if (parsed && parsed.errors.length > 0) {
      parsed.options.noEmit = true;
    }
    return parsed;
  };
  return host;
}
export {
  forceNoEmitOnSolutionBuilderHost
};
